### PR TITLE
chore: improve the build doc step in the release action

### PR DESCRIPTION
**Pull request by RomanBredehoft**
during the release process, links are not checked anymore  (in order to avoid temporary connection errors) + the documentation is built with reference builds only

### CODE REVIEW

File: .github/workflows/continuous-integration.yaml [hunk 1]
```diff
@@ -562,22 +564,17 @@ jobs:
...
-          make docs
-
-      # The changelog is generated by considering all commits from the latest stable previous 
-      # version (not a release candidate) up to the new upcoming version
-      # This may change in the future
-      # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3909
-      - name: Generate release changelog
-        id: changelog
-        if: ${{ fromJSON(env.IS_RELEASE) && steps.install-deps.outcome == 'success' && !cancelled() }}
+          make docs_no_links
```
Comment (RomanBredehoft) [inline .github/workflows/continuous-integration.yaml:567]: don't check links by default when building docs

File: .github/workflows/continuous-integration.yaml [hunk 2]
```diff
@@ -562,22 +564,17 @@ jobs:
...
+      
+      # Do not check links during the release process in order to avoid temporary connection errors
+      - name: Check links
+        id: check_links
+        if: |
+          !fromJSON(env.IS_RELEASE)
```
Comment (RomanBredehoft) [inline .github/workflows/continuous-integration.yaml:573]: only check links when we are not releasing

File: .github/workflows/continuous-integration.yaml [hunk 3]
```diff
@@ -589,54 +586,86 @@ jobs:
...
-            echo "Build docs step: ${{ steps.build-docs.outcome }}"
             echo "Determinism step: ${{ steps.determinism.outcome }}"
+            echo "Build docs step: ${{ steps.build-docs.outcome }}"
+            echo "Check links step: ${{ steps.check_links.outcome }}"
             exit 1
           fi
 
+      # Tar the docs for releases with the reference build only
       # Taring the docs allows for much faster upload speed (from ~3min worst case to ~2s best case)
       - name: Tar docs artifacts
```
Comment (RomanBredehoft) [inline .github/workflows/continuous-integration.yaml:606]: the following is basically avoiding doc management needed for the release when we are not running the CI for a release on the reference build

File: .github/workflows/continuous-integration.yaml [hunk 4]
```diff
@@ -551,7 +552,8 @@ jobs:
         id: build-docs
         if: |
           (
-            steps.changed-files-in-pr.outcome == 'skipped' 
+            (fromJSON(env.IS_RELEASE) && fromJSON(env.IS_REF_BUILD))
```
Comment (RomanBredehoft) [inline .github/workflows/continuous-integration.yaml:555]: for releases, only build docs if the CI is on the reference build (python 3.8)

File: Makefile [hunk 5]
```diff
@@ -386,8 +386,9 @@ docker_clean_volumes:
 .PHONY: docker_cv # Docker clean volumes
 docker_cv: docker_clean_volumes
 
-.PHONY: docs # Build docs
-docs: clean_docs check_docs_dollars
+
+.PHONY: docs_no_links # Build docs
+docs_no_links: clean_docs check_docs_dollars
```
Comment (RomanBredehoft) [inline Makefile:391]: added the new command that build docs without checking links

File: .github/workflows/continuous-integration.yaml [hunk 6]
```diff
@@ -92,7 +92,7 @@ env:
   GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
   AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
- 
+  FAILED_TESTS_ARE_FLAKY: "false"
```
Comment (fd0r) [inline .github/workflows/continuous-integration.yaml:95]: What is this variable?

File: .github/workflows/continuous-integration.yaml [hunk 7]
```diff
@@ -551,7 +552,8 @@ jobs:
         id: build-docs
         if: |
           (
-            steps.changed-files-in-pr.outcome == 'skipped' 
+            (fromJSON(env.IS_RELEASE) && fromJSON(env.IS_REF_BUILD))
+            || steps.changed-files-in-pr.outcome == 'skipped'
```
Comment (fd0r) [inline .github/workflows/continuous-integration.yaml:556]: Why is this needed?

File: .github/workflows/continuous-integration.yaml (hunk 2 above)
Comment (fd0r) [reply .github/workflows/continuous-integration.yaml:573]: That's great!

**Review by fd0r — APPROVED** [review]
Some questions to better understand some modifications but overall looks good to me.
Great to remove another release blocker, thanks @RomanBredehoft ! 🙏🏼

File: .github/workflows/continuous-integration.yaml (hunk 7 above)
Comment (RomanBredehoft) [reply .github/workflows/continuous-integration.yaml:556]: it was already here before (see above), it just moved (but basically we avoid building docs in PRs if no documentation files changed)
